conans/client/generators/virtualenv.py

```diff
@@ -17,36 +17,52 @@ def __init__(self, conanfile):
+    def get_setenv_variables(self):
+        """Returns a list of dictionaries with the new values for the environment,
+        the respective separator if needed, and their representation for the
+        activate script.
+        """
         ret = []
         for name, value in self.env.items():
             if platform.system() == "Windows":
                 if isinstance(value, list):
                     value = os.pathsep.join(value)
-                    ret.append(command_set + ' ' + name + '=' + value + ';%' + name + '%')
+                    ret.append({'name': name, 'value': value, 'separator': os.pathsep,
+                                'scriptvalue': os.pathsep.join([value, '%' + name + '%'])})
                 else:
                     # CMake crashes with quotes in CXX and CC, it seems
                     # that windows doesn't need the quotes, it handles right
                     # the spaces: #1169
-                    ret.append(command_set + ' ' + name + '=' + value)
+                    ret.append({'name': name, 'value': value, 'scriptvalue': value})
             else:
                 if isinstance(value, list):
                     if name in self.append_with_spaces:
                         # Variables joined with spaces look like: CPPFLAGS="one two three"
                         value = " ".join(value)
-                        ret.append(name + '="' + value + '%s$' % " " + name + '"')
+                        ret.append({'name': name, 'value': value, 'separator': ' ',
+                                    'scriptvalue': '"' + ' '.join([value, '$' + name]) + '"'})
                     else:
-                        # Variables joined with : or ; (os.pathset) look like: PATH="one path":"two paths"
-                        value = os.pathsep.join(['"%s"' % val for val in value])
-                        ret.append(name + '=' + value + '%s$' % os.pathsep + name)
-
-                    # Standard UNIX "sh" does not allow "export VAR=xxx" on one line
-                    # So for portability reasons we split into two commands
-                    ret.append(command_set + ' ' + name)
+                        # To keep current beahviour, variables in the script are
+                        # joined os.pathset and quotes: PATH="one path":"two paths"
+                        scriptvalue = os.pathsep.join(['"%s"' % val for val in value])
+                        value = os.pathsep.join(value)
+                        ret.append({'name': name, 'value': value, 'separator': os.pathsep,
+                                    'scriptvalue': os.pathsep.join([scriptvalue, '$' + name])})
                 else:
-                    ret.append(name + '="' + value + '"')
-                    ret.append(command_set + ' ' + name)
+                    ret.append({'name': name, 'value': value, 'scriptvalue': '"' + value + '"'})
+        return ret
+
+    def _get_setenv_variables_commands(self):
```

Comment:
Could you please clarify the changes of this method? It is not clear at all to the scope of this PR.